auth-ui/src/main/java/com/google/android/horologist/auth/ui/common/screens/SignInPromptScreen.kt

```
@@ -0,0 +1,135 @@
+import androidx.lifecycle.viewmodel.compose.viewModel
+import androidx.wear.compose.material.AutoCenteringParams
+import androidx.wear.compose.material.CircularProgressIndicator
+import androidx.wear.compose.material.MaterialTheme
+import androidx.wear.compose.material.ScalingLazyColumn
+import androidx.wear.compose.material.ScalingLazyColumnDefaults
+import androidx.wear.compose.material.ScalingLazyListScope
+import androidx.wear.compose.material.ScalingLazyListState
+import androidx.wear.compose.material.ScalingParams
+import androidx.wear.compose.material.Text
+import androidx.wear.compose.material.rememberScalingLazyListState
+import com.google.android.horologist.auth.composables.R
+import com.google.android.horologist.auth.composables.dialogs.SignedInConfirmationDialog
+import com.google.android.horologist.auth.ui.ExperimentalHorologistAuthUiApi
+import com.google.android.horologist.base.ui.components.Title
+import com.google.android.horologist.compose.focus.RequestFocusWhenActive
+import com.google.android.horologist.compose.rotaryinput.rotaryWithFling
+
+@ExperimentalHorologistAuthUiApi
+@Composable
+public fun SignInPromptScreen(
+    message: String,
+    onAlreadySignedIn: () -> Unit,
+    modifier: Modifier = Modifier,
+    title: String = stringResource(id = R.string.horologist_signin_prompt_title),
+    viewModel: SignInPromptViewModel = viewModel(),
+    scalingLazyListState: ScalingLazyListState = rememberScalingLazyListState(),
+    scalingParams: ScalingParams = ScalingLazyColumnDefaults.scalingParams(),
+    autoCentering: AutoCenteringParams? = AutoCenteringParams(),
+    content: ScalingLazyListScope.() -> Unit
+) {
+    var executedOnce by rememberSaveable { mutableStateOf(false) }
+    val focusRequester = remember { FocusRequester() }
+    val state by viewModel.uiState.collectAsStateWithLifecycle()
+
+    when (state) {
+        SignInPromptScreenState.Idle -> {
+            SideEffect {
```

Should there be UI here? or meant to be blank while it transitions/

---

In theory it would never stay in this Idle state, but given the VM is open at the moment, it's out of our control. I would be tempted to throw an error and let it crash as it's a developer usage issue, or it could be "robust" and either just leave it blank or display the same view as "loading" state. Thoughts? Otherwise I might just go with the option to display the view for "loading" state.

---

Yep. I guess minimise flicker in real world usage.